tests/transaction_receipt_idx_test.go

```diff
@@ -0,0 +1,92 @@
+package tests
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/0xsoniclabs/sonic/gossip/contract/driverauth100"
+	"github.com/0xsoniclabs/sonic/opera"
+	"github.com/0xsoniclabs/sonic/opera/contracts/driverauth"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReceipt_InternalTransactionsDoNotChangeReceiptIndex(t *testing.T) {
+	upgrades := opera.GetSonicUpgrades()
+	net := StartIntegrationTestNetWithJsonGenesis(t, IntegrationTestNetOptions{
+		Upgrades: &upgrades,
+	})
+
+	client, err := net.GetClient()
+	require.NoError(t, err)
+	defer client.Close()
+
+	chainId := net.GetChainId()
+	gasPrice, err := client.SuggestGasPrice(t.Context())
+	require.NoError(t, err)
+	sender := makeAccountWithBalance(t, net, big.NewInt(1e18))
+
+	numSimpleTxs := 10
+	transactions := prepareTransactions(t, chainId, sender, gasPrice, numSimpleTxs)
+
+	// Send first half of simple transactions
+	for i := range numSimpleTxs / 2 {
+		err = client.SendTransaction(t.Context(), transactions[i])
+		require.NoError(t, err)
+	}
+
+	// Send internal transaction (advance epoch)
+	contract, err := driverauth100.NewContract(driverauth.ContractAddress, client)
+	require.NoError(t, err)
+	txOpts, err := net.GetTransactOptions(&net.account)
+	require.NoError(t, err)
+	tx, err := contract.AdvanceEpochs(txOpts, big.NewInt(int64(1)))
+	require.NoError(t, err)
+	require.NotNil(t, tx)
+
+	// Send second half of simple transactions
+	for i := range numSimpleTxs / 2 {
+		err = client.SendTransaction(t.Context(), transactions[numSimpleTxs/2+i])
+		require.NoError(t, err)
+	}
+
+	// Wait for receipt of the internal transaction
+	receipt, err := net.GetReceipt(tx.Hash())
```

Comment:
Does the block containing the "AdvanceEpoch" transaction indeed containing the epoch-sealing transactions?

Usually, the epoch-sealing transactions are at the begin of a block. This would place them after the transaction signaling the epoch change.